### PR TITLE
fix(@angular/cli): Filter duplicate help statements

### LINKED ISSUE #12492: `--help` Outputs duplicate flags.
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

- [x] --help
```

### Versions

`node v10.11.0`
`npm v6.4.1`
`ng v6.2.3`
`Linux x64 with Ubuntu`

### Repro steps

`ng new --help`
or
`ng update --help`

### The log given by the failure

Example:
```
~❯ ng new --help
Creates a new directory and a new Angular app.
usage: ng new <name> [options]
arguments:
  name
    The name of the workspace.
options:
  --collection (-c)
    Schematics collection to use.
  --collection (-c)
    Schematics collection to use.
  --directory 
    The directory name to create the workspace in.
  --dryRun (-d)
    Run through without making any changes.
  --dryRun (-d)
    Run through without making any changes.
  --experimental-ivy 
    EXPERIMENTAL: Specifies whether to create a new application which uses the Ivy rendering engine.
  --force (-f)
    Forces overwriting of files.
  --force (-f)
    Forces overwriting of files.
  --inline-style (-s)
    Specifies if the style will be in the ts file.
  --inline-template (-t)
    Specifies if the template will be in the ts file.
  --new-project-root 
    The path where new projects will be created.
  --prefix (-p)
    The prefix to apply to generated selectors.
  --routing 
    Generates a routing module.
  --skip-git (-g)
    Skip initializing a git repository.
  --skip-install 
    Skip installing dependency packages.
  --skip-tests (-S)
    Skip creating spec files.
  --style 
    The file extension to be used for style files.
  --verbose (-v)
    Adds more details to output logging.
  --verbose (-v)
    Adds more details to output logging.
  --view-encapsulation 
    Specifies the view encapsulation strategy.
```
### Desired functionality

Remove the duplicate help statements.

### PR DESCRIPTION
**Pull request by rsarky**
Fixes #12492

### CODE REVIEW

File: packages/angular/cli/models/command.ts [hunk 1]
```diff
@@ -101,6 +101,9 @@ export abstract class Command<T extends BaseCommandOptions = BaseCommandOptions>
       this.logger.info(`options:`);
       opts
         .filter(o => !o.hidden)
+        .filter((val, index, self) => {
```
Comment (alan-agius4) [inline packages/angular/cli/models/command.ts]: Ideally we fix the schemas rather than filtering the out what’s duplicate.

Comment (rsarky) [reply packages/angular/cli/models/command.ts]: Hey. Thanks a lot for reviewing. 
I couldn't figure out what in the schema is causing the duplicate output

Comment (hansl) [reply packages/angular/cli/models/command.ts]: I actually think the proper fix might be in the [parseJsonSchemaToOptions](https://github.com/angular/angular-cli/blob/master/packages/angular/cli/utilities/json-schema.ts#L158) function. Having multiple times the same property isn't an error in JSON Schema world (as long as those properties match each others), but for the options you should only have one. So we should skips options we already found.

At that line, add a check for `if (options.some(o => o.name === name) { return; }` and you should be good. That way you fix the source of the problem and not the symptoms :)

Comment (rsarky) [reply packages/angular/cli/models/command.ts]: Understood.
I have updated the PR.

**Comment by rsarky** [comment]
Closing this as it is already handled in the latest release

**Pull request closed without merging**